feat: create ubuntu platform reference table

davidekete opened the pull request:
@edibotopic  closes issue https://github.com/canonical/open-documentation-academy/issues/198

edibotopic commented on ubuntu-desktop/reference/ubuntu-platform-reference.md.md:
```diff
@@ -0,0 +1,17 @@
+
+| **Name**              | **Target Platform**                                                        | **Features**                                                                                                                                                                                 | **Official Docs / Links**                                                                                 ...
+| --------------------- | -------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------ ...
+| **Ubuntu Desktop**    | Personal computers (laptops, desktops)                                     | <ul><li>GNOME-based GUI by default</li><li>Optimized for everyday computing</li><li>Extensive user-friendly ecosystem</li></ul>                                                                             | <ul><li>**Product page**: [ubuntu.com/desktop](https://ubuntu.com/desktop)</li></ul>       ...
```
```suggestion
| **Ubuntu Desktop**    | Personal computers (laptops, desktops)                                     | <ul><li>GNOME-based GUI by default</li><li>Optimized for everyday computing</li><li>Extensive range of software applications</li></ul>                                                                             | <ul><li>**Product page**: [ubuntu.com/desktop](https://ubuntu.com/desktop)</li></ul>                                                                                                                                              | <ul><li>General-purpose personal computing</li><li>Office productivity, browsing, media consumption</li></ul>                                             |
```

I think "extensive user-friendly ecosystem" was somewhat vague.

edibotopic commented on ubuntu-desktop/reference/ubuntu-platform-reference.md.md:
```diff
@@ -0,0 +1,17 @@
+| **Name**              | **Target Platform**                                                        | **Features**                                                                                                                                                                                 | **Official Docs / Links**                                                                                 ...
+| --------------------- | -------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------ ...
+| **Ubuntu Desktop**    | Personal computers (laptops, desktops)                                     | <ul><li>GNOME-based GUI by default</li><li>Optimized for everyday computing</li><li>Extensive user-friendly ecosystem</li></ul>                                                                             | <ul><li>**Product page**: [ubuntu.com/desktop](https://ubuntu.com/desktop)</li></ul>       ...
+| **Ubuntu Server**     | Physical or virtual servers, HPC clusters                                  | <ul><li>No GUI installed by default</li><li>Focus on stability and security</li><li>Long-term support releases available</li></ul>                                                                          | <ul><li>**Product page:** [ubuntu.com/server](https://ubuntu.com/server)</li><li>**Docs**: ...
```
```suggestion
| **Ubuntu Server**     | Physical or virtual servers, HPC clusters                                  | <ul><li>No GUI installed by default</li><li>Focus on stability and security</li><li>Long-term support releases available</li></ul>                                                                          | <ul><li>**Product page:** [ubuntu.com/server](https://ubuntu.com/server)</li><li>**Docs**: [https://documentation.ubuntu.com/server/](https://documentation.ubuntu.com/server/)</li></ul>                                         | <ul><li>Hosting web services, databases, or enterprise software</li><li>Infrastructure management</li></ul>                                               |
```

I think we need to be careful about suggesting there are features for certain platforms but not for others, when the truth is that they are available for both.

As one example, we suggest here that LTS releases are available for Server, but don't mention it for Desktop, Cloud, Core, WSL, etc. If there's something like this, which generally applies many cases, it might be worth mentioning at the top (before the table), something like, "Regardless of platform, Ubuntu offers Long term support releases [...]".

This is a general remark that you should keep in mind when revisiting this page.

edibotopic commented on ubuntu-desktop/reference/ubuntu-platform-reference.md.md:
```diff
@@ -0,0 +1,17 @@
+| **Ubuntu Cloud**      | Public, Private, Multi, & Hybrid clouds                                    | <ul><li>Preconfigured cloud images</li><li>Optimized for scalability and quick boot times</li><li>Integration with cloud services</li></ul>                                                                 | <ul><li>**Product page:** [ubuntu.com/cloud](https://ubuntu.com/cloud)</li></ul>           ...
+| **Ubuntu Core (IoT)** | IoT devices, embedded systems, robotics                                    | <ul><li>Entirely snap-based (transactional updates)</li><li>Minimal, lightweight OS</li><li>Automatic security patches</li></ul>                                                                            | <ul><li>**Product page**: [ubuntu.com/core](https://ubuntu.com/core)</li><li>**Docs:** [ht ...
+| **Ubuntu on WSL**     | Machines running the Windows OS                                            | <ul><li>Run Ubuntu inside Windows</li><li>Command-line tools and utilities available natively</li><li>Seamless developer workflow</li></ul>                                                                 | <ul><li>**Product page**: [ubuntu.com/wsl](https://ubuntu.com/wsl)</li><li>**Docs:** [http ...
+| **Ubuntu Pro**        | Enterprise, production environments requiring compliance                   | <ul><li>Extended security maintenance (ESM)</li><li>10 years of security updates</li><li>FIPS 140-2 certified components</li></ul>                                                                          | <ul><li>**Product page**: [ubuntu.com/pro](https://ubuntu.com/pro)</li><li>**Docs:** [http ...
```
Ubuntu Pro is an interesting outlier here.

All the others are Ubuntu distributions but Pro is a feature that can be used with some Ubuntu distributions, to enhance them in specific ways.

I would suggest removing it from the table, but perhaps mentioning in a note after the table that Ubuntu supports Ubuntu Pro on several platforms, including x, y and z. You could then summarise the information about Pro that you included in the table.

Somewhat related, Ubuntu Frame is a display server rather than a distribution.

Alternatively, you could have an additional column that explains what the entry is, for example: variant of Ubuntu distribution with real-time kernel; a display server for embedded devices.

These could be common confusions for people, so it might be valuable to be maximally clear.

davidekete replied:
So I took your suggestion and moved it from the table to the notes section. I think removing ubuntu frame would be more appropriate in this situation, less overhead that creating an entirely new column to justify the entry.

edibotopic commented on ubuntu-desktop/reference/ubuntu-platform-reference.md.md:
```diff
@@ -0,0 +1,17 @@
+
+### Notes
+
+- **Flavors**: Beyond these main editions, there are official [Ubuntu flavors](https://ubuntu.com/desktop/flavours) (e.g., Kubuntu, Xubuntu, Ubuntu MATE, Ubuntu Budgie, Ubuntu Studio) each featuring a different desktop environment or specialized toolset.
```
```suggestion
- **Flavors**: There are also official [Ubuntu flavors](https://ubuntu.com/desktop/flavours) (e.g., Kubuntu, Xubuntu, Ubuntu MATE, Ubuntu Budgie, Ubuntu Studio), each featuring a different desktop environment or specialized toolset.
```

Simplified.

edibotopic commented on ubuntu-desktop/reference/ubuntu-platform-reference.md.md:
```diff
@@ -0,0 +1,17 @@
+# Ubuntu platform reference table
```
One thing to consider adding as a column is download size, or a typical range.

For example, the Ubuntu image for desktop is a lot larger than the image for WSL, which reflects the packages installed but also the constraints of the target platform and use case.

davidekete commented on ubuntu-desktop/reference/ubuntu-platform-reference.md.md:
```diff
@@ -0,0 +1,17 @@
+# Ubuntu platform reference table
+The table below provides an overview of Ubuntu platforms, their features, use cases, and official links.
```
Alright, the changes look good.

davidekete commented:
Hi Shane, I think I have addressed most of your comments. Please confirm that everything looks good so I can make I commit that fixes the markdown formatting. I think it got broken at some point.

edibotopic reviewed (approved):
Thanks @davidekete -- I will give preliminary approval.

Few remarks:

* You decided to keep Frame in the table?
* Did you not think adding download size was a good idea?
* If you could fix up the markdown table formatting, that would be great

Thanks again.

davidekete commented:
> Thanks @davidekete -- I will give preliminary approval.
> 
> Few remarks:
> 
> * You decided to keep Frame in the table?
> * Did you not think adding download size was a good idea?
> * If you could fix up the markdown table formatting, that would be great
> 
> Thanks again.

Hi @edibotopic, I removed Ubuntu Frame in my last commits and fixed the formatting. I missed your suggestion about the sizes column earlier. Upon reviewing––wouldn't that column be hard to maintain considering the fact that the download sizes may vary across releases? also it's not information a user can retain easily. What were your thoughts on adding it?

edibotopic reviewed (approved):
LGTM.

Giving preliminary approval with one small change requested.

This documentation will not be made public for a while yet, but I will keep you updated on any major progress.
